omnigibson/envs/env_base.py

```diff
@@ -572,50 +636,51 @@ def reset(self):
         # Reset internal variables
         self._reset_variables()
 
-        # Run a single simulator step to make sure we can grab updated observations
-        og.sim.step()
-
-        # Grab and return observations
-        obs, _ = self.get_obs()
-
-        if self._loaded:
-            # Sanity check to make sure received observations match expected observation space
-            check_obs = recursively_generate_compatible_dict(dic=obs)
-            if not self.observation_space.contains(check_obs):
-                exp_obs = dict()
-                for key, value in recursively_generate_flat_dict(dic=self.observation_space).items():
-                    exp_obs[key] = ("obs_space", key, value.dtype, value.shape)
-                real_obs = dict()
-                for key, value in recursively_generate_flat_dict(dic=check_obs).items():
-                    if isinstance(value, np.ndarray):
-                        real_obs[key] = ("obs", key, value.dtype, value.shape)
-                    else:
-                        real_obs[key] = ("obs", key, type(value), "()")
-
-                exp_keys = set(exp_obs.keys())
-                real_keys = set(real_obs.keys())
-                shared_keys = exp_keys.intersection(real_keys)
-                missing_keys = exp_keys - real_keys
-                extra_keys = real_keys - exp_keys
-
-                if missing_keys:
-                    log.error("MISSING OBSERVATION KEYS:")
-                    log.error(missing_keys)
-                if extra_keys:
-                    log.error("EXTRA OBSERVATION KEYS:")
-                    log.error(extra_keys)
-
-                mismatched_keys = []
-                for k in shared_keys:
-                    if exp_obs[k][2:] != real_obs[k][2:]:  # Compare dtypes and shapes
-                        mismatched_keys.append(k)
-                        log.error(f"MISMATCHED OBSERVATION FOR KEY '{k}':")
-                        log.error(f"Expected: {exp_obs[k]}")
-                        log.error(f"Received: {real_obs[k]}")
-
-                raise ValueError("Observation space does not match returned observations!")
-
-        return obs
+        if get_obs:
```

Comment:
when would `get_obs` be False? I thought gym interface requires us to return obs from reset.

Reply:
It seems like obs and info are required here, looping @cgokmen in here

Reply:
+1 to what eric said

Reply:
Gym does require this. This is for use in a non-gym context, e.g. when resetting multiple environments in a vector env. We don't render/step until all of the envs are reset, so any observations taken before they are all reset (and a render/step is done) are useless and we need to take observations after everything is reset, but the halfway-through get obs calls do still take a considerable amount of time. This allows us to skip those unnecessary observations. Again a result of the single-first environment design. To be fixed.